a10networks/a10-octavia
# STACK-1253 and STACK-1244 - Refactor and unison a10-octavia-neutron driver

## Description
Refactored neutron driver
Added create and delete port to a10-octavia-neutron driver
Merged together code for trunking + VRID + auto ve/vlan branches - code credits - @hthompson-a10  @ssrinivasan-a10 

## Jira Ticket
[STACK-1253](https://a10networks.atlassian.net/browse/STACK-1253)
[STACK-1244](https://a10networks.atlassian.net/browse/STACK-1244)

## Technical Approach
Used allowedaddresspair as parent driver to write custom a10-octavia-neutron driver so that we can enhance the functionalities supported by the neutron driver.
Added create and delete port functions, used feature/trunking branch to add trunking solutions to the master.





## Test Cases
- Booted a vThunder with normal approach
- Booted a vThunder with rack approach
- Created a port for VRID member creation 
- Deleted a port for VRID member deletion

## Manual Testing
- Create a LB, check network driver was called
- create a LB on rack, check network driver was not called
- create a member with VRID, check create port called
- change VRID, check delete port was called
